### PR TITLE
[Recure]: added local storage support

### PR DESCRIPTION
**Pull request by Abnormaltype**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -97,15 +103,47 @@ async function getPayload(
 }
 
 function isReadyToSend(): boolean {
-  const readyToSend: string | undefined = Cookies.get(sendingEventsBlocked);
+  const cReadyToSend: string | undefined = Cookies.get(sendingEventsBlocked);
```
Comment (ArtemTopolskyi) [inline src/index.ts:106]:
```suggestion
  const cReadyToSend?: string = Cookies.get(sendingEventsBlocked);
```

**Review by ArtemTopolskyi — COMMENTED** [review]
Lets structure code in more readable way: create contants for reusable numbers.
Also you can move localStorage related logic to separate LocalStorageManager or something

File: src/index.ts (hunk 1 above)
Comment (Abnormaltype) [reply src/index.ts:106]: image

File: src/index.ts [hunk 2]
```diff
@@ -96,16 +104,50 @@ async function getPayload(
   };
 }
 
+function setNewDateToLocalStorageAndCookies(expirationTime: Date): void {
```
[the remark this answers is not in the document]
Comment (Abnormaltype) [reply src/index.ts]: done

File: src/index.ts (hunk 1 above)
Comment (danylott) [reply src/index.ts:134]: Cookies<str>?

File: src/index.ts [hunk 3]
```diff
@@ -4,9 +4,11 @@ import * as Cookies from 'js-cookie';
 
 const apiKeyUrl: string = 'https://api.recure.ai/api/event_handler/get_api_key/';
 const eventHandlerUrl: string = 'https://api.recure.ai/api/event_handler/';
-const daysToExpire: number = 60;
+const daysToExpire: number = 120;
 const minutesToExpire: number = 5;
 const sendingEventsBlocked: string = 'recureSendingEventsBlocked';
+const toSeconds: number = 1000;
+const toMinutes: number = 60;
```
Comment (danylott) [inline src/index.ts:11]: What to Seconds?
`secondsToMiliseconds` you mean?

File: src/index.ts [hunk 4]
```diff
@@ -53,10 +55,15 @@ async function getApiKey(url: string, projectApiKey: string): Promise<string> {
 }
 
 async function getOrSetVisitorId(cookieName: string, projectApiKey: string): Promise<string> {
-  const visitorId: string | undefined = Cookies.get(cookieName);
+  const cVisitorId: string | undefined = Cookies.get(cookieName);
+  const lSVisitorId: string | null = localStorage.getItem(cookieName);
 
-  if (visitorId !== '' && visitorId !== undefined) {
-    return visitorId;
+  if (cVisitorId && cVisitorId) {
```
Comment (danylott) [inline src/index.ts:61]: Why you repeating twice `cVisitorId` here?

File: src/index.ts [hunk 5]
```diff
@@ -53,10 +55,15 @@ async function getApiKey(url: string, projectApiKey: string): Promise<string> {
...
+    return cVisitorId;
+  }
+
+  if (lSVisitorId && lSVisitorId) {
```
Comment (danylott) [inline src/index.ts:65]: same

**Review by danylott — APPROVED** [review]

File: src/index.ts [hunk 6]
```diff
@@ -53,10 +55,22 @@ async function getApiKey(url: string, projectApiKey: string): Promise<string> {
...
+  if (cVisitorId) {
+    if (!lSVisitorId) {
+      setToLocalStorage(cookieName, cVisitorId);
+    }
+
+    return cVisitorId;
+  }
+
+  if (lSVisitorId) {
+    if (!cVisitorId) {
```
Comment (danylott) [inline src/index.ts:70]: Use &, but not double if

**Review by danylott — APPROVED** [review]

**Pull request merged**